### PR TITLE
update README

### PR DESCRIPTION
**Pull request by zz-jason**
@coocood @disksing PTAL

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,26 +1,51 @@
...
-We generate all go code in pkg folder and rust in src folder.
+## Generate go and rust codes
 
-+ `make`
+```sh
+make
+```
 
-### Update the dependent projects.
+NOTE: Do not forget to update the dependent projects !
```
Comment (Hoverbear) [inline README.md]: The `s` and `!` should be together, no space.

File: README.md [hunk 2]
```diff
@@ -1,26 +1,51 @@
...
+```sh
+go get -u github.com/gogo/protobuf/protoc-gen-gofast
+cd $GOPATH/src/github.com/gogo/protobuf
+git checkout v0.5
+rm $GOPATH/bin/protoc-gen-gofast
+go get github.com/gogo/protobuf/protoc-gen-gofast
+```
 
-We generate all go code in pkg folder and rust in src folder.
+## Generate go and rust codes
```
Comment (Hoverbear) [inline README.md]: `Generate the Go and Rust code`

File: README.md [hunk 3]
```diff
@@ -1,26 +1,51 @@
...
 
-# Usage
+## PreRequests
+
+### 1. Install [google/protobuf](https://github.com/google/protobuf)
+
+We use protoc 3.5.1:
+
+```sh
+# install protoc
```
Comment (disksing) [inline README.md]: How about download a release instead?

**Comment by coocood** [comment]
LGTM

**Comment by zz-jason** [comment]
@disksing @Hoverbear @AndreMouche PTAL, thanks

**Comment by shenli** [comment]
Well done!

**Pull request merged**